backend/drm/session.c

```diff
@@ -0,0 +1,219 @@
+#define _POSIX_C_SOURCE 200809L
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <systemd/sd-bus.h>
+#include <systemd/sd-login.h>
+#include <unistd.h>
+#include <sys/sysmacros.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "backend/drm/backend.h"
+#include "backend/drm/session.h"
+#include "common/log.h"
+
+int wlr_session_take_device(struct wlr_session *restrict session,
+		const char *restrict path,
+		bool *restrict paused_out)
+{
+	int ret;
+	int fd = -1;
+	sd_bus_message *msg = NULL;
+	sd_bus_error error = SD_BUS_ERROR_NULL;
+
+	struct stat st;
+	if (stat(path, &st) < 0) {
+		wlr_log(L_ERROR, "Failed to stat '%s'", path);
+		return -1;
+	}
+
+	ret = sd_bus_call_method(session->bus,
+				 "org.freedesktop.login1",
```

Comment:
Style: don't indent with respect to the `(`, just add a single tab on subsequent lines. Also merge parameters until you hit 80 columns before breaking.